### PR TITLE
MON-3126: osde2e: add observability-operator test harness job

### PR DESCRIPTION
**Pull request by jan--f**

### CODE REVIEW

File: ci-operator/config/rhobs/observability-operator/rhobs-observability-operator-main__periodic.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      memory: 200Mi
+tests:
+- as: e2e-test-harness
+  commands: |
+    export REPORT_DIR="$ARTIFACT_DIR"
+    export CONFIGS="rosa,stage,test-harness"
+    export ROSA_ENV="stage"
+    export ROSA_STS="true"
+    export ROSA_AWS_REGION="random"
+    export TEST_HARNESSES="quay.io/rhobs/observability-operator-test-harness"
```
Comment (lihongyan1) [inline ci-operator/config/rhobs/observability-operator/rhobs-observability-operator-main__periodic.yaml:21]: @jan--f @sthaha , not sure if this is expected
quay.io/rhobs/observability-operator-test-harness is not available
quay.io/sthaha/observability-operator-test-harness is available

File: ci-operator/config/rhobs/observability-operator/rhobs-observability-operator-main__periodic.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    export CHANNEL="stable"
+    export SECRET_LOCATIONS="/usr/local/osde2e-common,/usr/local/osde2e-credentials,/usr/local/osde2e-rosa-stage"
+    /osde2e test --secret-locations ${SECRET_LOCATIONS} --configs ${CONFIGS}
```
Comment (lihongyan1) [inline ci-operator/config/rhobs/observability-operator/rhobs-observability-operator-main__periodic.yaml:24]: According to my test, locally run test-harness with the following scripts works, should change the commands accordingly
```
#!/usr/bin/env bash
set -x
export AWS_SECRET_ACCESS_KEY="$"
export AWS_ACCESS_KEY_ID="$"
export OCM_TOKEN="$" 
CLUSTER_ID="23u83n9fp9sldmm0eq00f084hkobu2g2"   \
TEST_HARNESSES="quay.io/sthaha/observability-operator-test-harness:latest" \
ROSA_ENV="stage" \
ROSA_AWS_REGION="us-west-2" \
CLOUD_PROVIDER_REGION="us-west-2" \
POLLING_TIMEOUT="1800" \
./out/osde2e test \
--configs rosa,stage,test-harness \
--must-gather true \
--skip-health-check true \
--skip-destroy-cluster true 
```

File: ci-operator/config/rhobs/observability-operator/rhobs-observability-operator-main__periodic.yaml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+      memory: 200Mi
+tests:
+- as: e2e-test-harness
+  commands: |
+    export REPORT_DIR="$ARTIFACT_DIR"
+    export CONFIGS="rosa,stage,test-harness"
+    export ROSA_ENV="stage"
+    export ROSA_STS="true"
+    export ROSA_AWS_REGION="random"
+    export TEST_HARNESSES="quay.io/rhobs/observability-operator-test-harness:latest"
```
Comment (sthaha) [inline ci-operator/config/rhobs/observability-operator/rhobs-observability-operator-main__periodic.yaml:21]: @lihongyan1 Should we rebuild and push this image?

Comment (lihongyan1) [reply ci-operator/config/rhobs/observability-operator/rhobs-observability-operator-main__periodic.yaml:21]: I executed the following and get error, seems I have no permission to do the push
`make osd-e2e-test-push IMAGE_BASE="quay.io/rhobs/observability-operator"`

**Comment by lihongyan1** [comment]
@ritmun From the code, I see no OCP version is provided, when the job is executed regularly/daily, what ocp version will be covered?
@jan--f Per osde2e doc, the job will run on aws ROSA stage,  if want to run test on other env, such as osd/gcp, osd/azure, new job needs to be created, so the job name better indicate such kind of information. If I am wrong, correct me @ritmun

**Comment by jan--f** [comment]
ok this test works now, previous failures where due to an outdated test harness image. We will add an automation to update said image regularly, this is tracked in https://github.com/rhobs/observability-operator/issues/298.

**Comment by jan--f** [comment]
/remove-lifecycle stale

Getting our aws account takes longer then expected :(

File: ci-operator/jobs/rhobs/observability-operator/rhobs-observability-operator-main-periodics.yaml [hunk 4]
```diff
@@ -54,6 +58,9 @@ periodics:
         readOnly: true
     serviceAccountName: ci-operator
     volumes:
+    - name: aws-qe-obo
+      secret:
+        secretName: aws-qe-obo
```
Comment (lihongyan1) [inline ci-operator/jobs/rhobs/observability-operator/rhobs-observability-operator-main-periodics.yaml:63]: need `namespace: test-credentials` for the secret

Comment (jan--f) [reply ci-operator/jobs/rhobs/observability-operator/rhobs-observability-operator-main-periodics.yaml:63]: I added that to the config spec, but the `jobs` make target removes that.

**Comment by jan--f** [comment]
/pj-rehearse periodic-ci-rhobs-observability-operator-main-periodic-e2e-test-harness

**Comment by jan--f** [comment]
/pj-rehearse periodic-ci-rhobs-observability-operator-main-periodic-e2e-test-harness

I updated the test harness image.

**Comment by jan--f** [comment]
@ritmun can this go ahead?

**Pull request merged**